core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -752,11 +752,10 @@ private[spark] class Executor(
         if (currentTimeStamp < timestamp) {
           logInfo("Fetching " + name + " with timestamp " + timestamp)
           // Fetch file with useCache mode, close cache for local mode.
-          Utils.fetchFile(name, new File(SparkFiles.getRootDirectory()), conf,
-            env.securityManager, hadoopConf, timestamp, useCache = !isLocal)
+          val url = Utils.fetchFile(name, new File(SparkFiles.getRootDirectory()), conf,
+            env.securityManager, hadoopConf, timestamp, useCache = !isLocal,
+            conf.getBoolean("spark.jars.withDecoratedName", false)).toURI.toURL
           currentJars(name) = timestamp
-          // Add it to our class loader
-          val url = new File(SparkFiles.getRootDirectory(), localName).toURI.toURL
```

Comment:
shall we still need the `localName` here?

Reply:
Well, I think we don't need it . The `File` returned by `Utils.fetchFile` is the local file. We don't need `localName` to initialize the local file here.